remove caffe2/core dependency from ATen/core/jit_type.h

ljk53 opened the pull request:
Summary: This include doesn't seem to be needed. Remove it to simplify mobile build dependency.

Differential Revision: D16088224

ljk53 commented on torch/csrc/jit/passes/guard_elimination.cpp:
```diff
@@ -1,3 +1,4 @@
+#include <caffe2/core/common.h>
```
@Krovatkin any reason we need use "caffe2::make_unique" below?
Seems some places use "torch::make_unique", some use "c10::guts::make_unique" and some use "std::make_unique" directly. Is there a guideline to follow?

dzhulgakov replied:
std::make_unique is c++14 only. Just change it to c10::guts::make_unique or torch::make_unique and call it a day

Krovatkin replied:
what Dima said. I needed to use unique ownership here since profiling embeds raw pointers into a graph and I just wanted to make sure we always know that they won't be accessed in some arbitrary way.

ljk53 committed:
remove caffe2/core dependency from ATen/core/jit_type.h (#22441)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22441

This include doesn't seem to be needed. Remove it to simplify mobile build dependency.

Reviewed By: dreiss

Differential Revision: D16088224

fbshipit-source-id: 0397918222e18eae25f167ec1ecde369a589e50a
